savant/deepstream/meta/object.py

```diff
@@ -75,20 +75,11 @@ def __init__(
         self.track_id = track_id
         self.parent = parent
 
-        if (
-            isinstance(parent, ObjectMeta)
-            and isinstance(parent.object_meta_impl, _NvDsObjectMetaImpl)
-            or parent is None
-        ):
-            pyds.nvds_add_obj_meta_to_frame(
-                self._frame_meta.frame_meta,
-                self.ds_object_meta,
-                parent.object_meta_impl.ds_object_meta if parent is not None else None,
-            )
-        else:
-            raise MetaValueError(
-                f'parent is `{type(parent)}` type, expected `{self.__class__.__name__}`'
-            )
+        pyds.nvds_add_obj_meta_to_frame(
+            self._frame_meta.frame_meta,
+            self.ds_object_meta,
+            parent.object_meta_impl.ds_object_meta if parent is not None else None,
```

Comment:
Why did you remove the check? The user can simply create an ObjectMeta and specify it as a parent. When an ObjectMeta is created, it is not immediately added to the DS meta and therefore its object_meta_impl will be None. Eventually object_meta_impl will be None and there will be an error.

Reply:
Because a line above `self.parent = parent` (parent setter) checks the same.

Reply:
Because a line above `self.parent = parent` (parent setter) checks the same.